tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/EditVenue/DetailsTests.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        [Fact]
+        public async Task Get_ValidRequestNoExistingFormFlowInstance_RendersExpectedOutputFromDatabase()
+        {
+            // Arrange
+            var providerId = await TestData.CreateProvider();
+            var venueId = await TestData.CreateVenue(
+                providerId,
+                venueName: "Test Venue",
+                email: "test-venue@provider.com",
+                telephone: "01234 567890",
+                website: "provider.com/test-venue",
+                addressLine1: "Test Venue line 1",
+                town: "Town",
+                postcode: "AB1 2DE");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"venues/{venueId}");
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var doc = await response.GetDocument();
+            Assert.Equal("Test Venue", doc.GetSummaryListValueWithKey("Location name"));
+            Assert.Equal(
+                "Test Venue line 1\nTown\nAB1 2DE",
+                doc.GetSummaryListValueWithKey("Address"));
+            Assert.Equal("test-venue@provider.com", doc.GetSummaryListValueWithKey("Email"));
+            Assert.Equal("01234 567890", doc.GetSummaryListValueWithKey("Phone"));
+            Assert.Equal("provider.com/test-venue", doc.GetSummaryListValueWithKey("Website"));
+        }
+
+        [Fact]
+        public async Task Get_ValidRequestWithExistingFormFlowInstance_RendersExpectedOutputFromState()
+        {
+            // Arrange
+            var providerId = await TestData.CreateProvider();
+            var venueId = await TestData.CreateVenue(providerId);
+
+            var formFlowInstance = await CreateFormFlowInstance(venueId);
+            formFlowInstance.UpdateState(state =>
+            {
+                state.Name = "Updated name";
+                state.Email = "updated@provider.com";
+                state.PhoneNumber = "02345 678901";
+                state.Website = "updated-provider.com";
+                state.AddressLine1 = "Updated line 1";
+                state.AddressLine2 = "Updated line 2";
+                state.Town = "Updated town";
+                state.County = "Updated county";
+                state.Postcode = "UP1 D8D";
+                state.Latitude = 42;
+                state.Longitude = 42;
+                state.NewAddressIsOutsideOfEngland = false;
+            });
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"venues/{venueId}");
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var doc = await response.GetDocument();
+            Assert.Equal("Updated name", doc.GetSummaryListValueWithKey("Location name"));
```

Comment:
I prefer to use a shared `const string` for the test values as it makes it clear to the reader and compiler that the values for setup and assertion are both required to be the same.

Reply:
> * The tests seem quite repetitive, lots of similar Forbidden/NotFound tests etc, can we DRY this? I'm worried that when we want to change things the sheer volume of test code will slow us down.

I've not found a simple way of doing this. From doing this style of tests for a number of years I haven't yet hit issues with the repition.